scripts/main.component.ts

```diff
@@ -20,18 +21,36 @@ import { REPLComponent, REPLDelegate } from "./repl.component"
   templateUrl: "../html/main.component.html",
   providers: [MenuService, PluginService]
 })
-export class MainComponent implements OnInit, MenuDelegate, REPLDelegate {
+export class MainComponent implements OnInit, MenuDelegate, REPLDelegate, EditorDelegate {
   constructor(private menu: MenuService, private pluginService: PluginService) {}
+  
+  selectedElements : Set<DrawableGraph | DrawableNode | DrawableEdge> =
+    new Set<DrawableGraph | DrawableNode | DrawableEdge>();
+  selectElement(element : DrawableGraph | DrawableNode | DrawableEdge) {
```

Comment:
I still think we should make the editor do the book keeping, and only inform the delegate of changes. For instance, what if there's no delegate set? Or if the delegate does something weird. Then the graph editor wouldn't work.